tmd/Tree/methods.py

```diff
@@ -87,11 +90,19 @@ def get_segments(self):
     return seg_list
 
 
-def get_segment_lengths(tree):
+def get_segment_lengths(tree, seg_nums=None):
     '''Returns segment lengths
+    Input
+    ------
+    tree: tmd tree
+    seg_nums: segment numbers to consider
     '''
-    seg_len = np.zeros(size(tree) - 1)
-    segs = tree.get_segments()
+    if not seg_nums:
+        seg_nums = range(1, size(tree))
+
+    seg_len = np.zeros(len(seg_nums))
```

Comment:
A shorter way to avoid the extra allocations is:
```python

return np.fromiter(
    (_rd(p_beg, p_end) for p_beg, p_end in tree.get_segments(seg_ids)),
     dtype=np.float32
)
```
Or if you want super fancy:
```python
from itertools import starmap
return np.fromiter(starmap(_rd, tree.get_segments(seg_ids)), dtype=np.float32)
```

Reply:
Nice, I didn't know `starmap` :)